src/webgpu/api/operation/rendering/blending.spec.ts

```diff
@@ -7,7 +8,269 @@ TODO:
 - ?
 `;
 
+import { params, poptions } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
+
+const kBlendFactors: GPUBlendFactor[] = [
+  'zero',
+  'one',
+  'src-color',
+  'one-minus-src-color',
+  'src-alpha',
+  'one-minus-src-alpha',
+  'dst-color',
+  'one-minus-dst-color',
+  'dst-alpha',
+  'one-minus-dst-alpha',
+  'src-alpha-saturated',
+  'blend-color',
+  'one-minus-blend-color',
+];
+
+const kBlendOperations: GPUBlendOperation[] = [
+  'add', //
+  'subtract',
+  'reverse-subtract',
+  'min',
+  'max',
+];
+
+function mapColor(
+  col: GPUColorDict,
+  f: (v: number, k: keyof GPUColorDict) => number
+): GPUColorDict {
+  return {
+    r: f(col.r, 'r'),
+    g: f(col.g, 'g'),
+    b: f(col.b, 'b'),
+    a: f(col.a, 'a'),
+  };
+}
+
+function computeBlendFactor(
+  src: GPUColorDict,
+  dst: GPUColorDict,
+  col: GPUColorDict,
+  factor: GPUBlendFactor
+): GPUColorDict {
+  switch (factor) {
+    case 'zero':
+      return { r: 0, g: 0, b: 0, a: 0 };
+    case 'one':
+      return { r: 1, g: 1, b: 1, a: 1 };
+    case 'src-color':
+      return { ...src };
+    case 'one-minus-src-color':
+      return mapColor(src, v => 1 - v);
+    case 'src-alpha':
+      return mapColor(src, () => src.a);
+    case 'one-minus-src-alpha':
+      return mapColor(src, () => 1 - src.a);
+    case 'dst-color':
+      return { ...dst };
+    case 'one-minus-dst-color':
+      return mapColor(dst, v => 1 - v);
+    case 'dst-alpha':
+      return mapColor(dst, () => dst.a);
+    case 'one-minus-dst-alpha':
+      return mapColor(dst, () => 1 - dst.a);
+    case 'src-alpha-saturated': {
+      const f = Math.min(src.a, 1 - src.a);
```

Comment:
```suggestion
      const f = Math.min(src.a, 1 - dst.a);
```
according to
https://docs.microsoft.com/en-us/windows/win32/api/d3d12/ne-d3d12-d3d12_blend